### ISSUE TITLE
Audio struck in Samsung Galaxy A52 - Android version 12

### ISSUE THREAD

**Issue by sowmiya5997** [pedroSG94/RootEncoder#1043]
Hi @pedroSG94 . In rtmp version 2.1.6, Audio got struck while streaming in **Samsung(Samsung Galaxy A52) android version 12** devices. What is the root cause for this issue and How should this to be fixed?

**Comment by pedroSG94** [comment]
What do you mean with struck?
Do you have this problem with multiple devices? which server are you using?

**Comment by sowmiya5997** [comment]
@pedroSG94 Audio is constantly breaking and choppy. Facing this issue on all the samsung devices which has Android version 12. We use AWS MediaLive for RTMP live transmission.

**Comment by pedroSG94** [comment]
For now, I did a commit for it:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/commit/3b1f26b3d65a03074a5d6a2828ec37885ce8d568

Please, compile my app example using OpenGl RTMP and add microphone mode in [this line](https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/blob/master/app/src/main/java/com/pedro/rtpstreamer/openglexample/OpenGlRtmpActivity.java#L131) BUFFER:
```java
rtmpCamera1.setMicrophoneMode(MicrophoneMode.BUFFER);
```

Let me know if it is working for you.

**Comment by sowmiya5997** [comment]
I did the same as you said. Still the issue is there.

**Comment by pedroSG94** [comment]
Ok, I was trying to reproduce your case but I need ask you multiple things:

- Do you mean Android 12 (API 31) or Android 12L (API 32)? Can you tell me all devices model tested?
- Did you have this problem with early Android version using that devices? (Android 11 or lower)
- Did you test in other device (No samsung) with the same API?
- If you record a local video using the library (startRecord), do you have audio problems?

**Comment by sowmiya5997** [comment]
@pedroSG94 
1. SM-G986B, SM-G998B, SM-G991B  -  these are the device models I tested. It has audio issue.
2. No, I am facing this audio issue only in Android 12 and 12L. 
3. Checked in pixel 5 (Android 12). It is wokring fine. 
4. I will check and let you know

**Comment by sowmiya5997** [comment]
Hi @pedroSG94.. Any update on audio issue?

**Comment by pedroSG94** [comment]
I'm on it. For now I did a change that seems to work on Android emulator:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/tree/a12audio
Compile app example in this branch and let me know if it is working on your samsung devices.

Test streaming and recording a local video and check it. Let me know the result.

**Comment by sowmiya5997** [comment]
@pedroSG94 Still the issue is there. Also we couldn't find the recorded video.

**Comment by pedroSG94** [comment]
Check again with the last commit in that branch and let me know if you can reproduce the issue.

If the problem persist. Let me know the exact model of A52:
A52 or A52 5G? exynos or qualcomn?

**Comment by sowmiya5997** [comment]
@pedroSG94 Checked it in the last commit. Still the issue is there.. FYR - Attching device details
Local recorded video - No audio issue
Live stream Video - Audio is not good
https://user-images.githubusercontent.com/61676699/152311373-844ba963-4c62-45a9-882f-ae4aedb9d2d7.jpg

https://user-images.githubusercontent.com/61676699/152311492-f2d9ed1c-dd2e-40bb-8b01-3254790b32ff.jpg

**Comment by pedroSG94** [comment]
> **Audio issue occurs only on Samsung android 12 devices**

Yes, it seems an issue only on Samsung android 12 / 12L and maybe only on YouTube and AWS servers.
 
Can you check using other server? This is easy to install and test:
https://github.com/ossrs/srs/wiki/v4_EN_Home#docker
https://github.com/iizukanao/node-rtsp-rtmp-server#installation-without-docker

I'm looking for the device with lowest price that can reproduce this problem to buy it.
For now Samsung A52 or Samsung S10. If you can find a device with Android 12 with similar price, let me know it.

**Comment by pedroSG94** [comment]
I have buy a samsung S10. It will take around 3 - 7 days to receive it and start tests.

**Comment by pedroSG94** [comment]
Hello,

Yesterday, I received a Samsung S10 and I did test for it. After few test, I discovered that the problem is related with the codec selected. I did a branch with the fix:
https://github.com/pedroSG94/rtmp-rtsp-stream-client-java/tree/samsungaudio
Can you test it and comfirm me that it is working in your devices?

**Comment by sowmiya5997** [comment]
Hey @pedroSG94 .. We tested with this branch. Seems audio issue in samsung is fixed. We could hear clear audio. Thanks a lot!!!

**Comment by sowmiya5997** [comment]
Hi @pedroSG94 .. Could you please update the library with this samsung audio fix asap?

**Comment by pedroSG94** [comment]
Hello,

I'm waiting for a user with similar problem to know if all is solved and then I can update the library.
For now, you can use this gradle:
```gradle
	        implementation 'com.github.pedroSG94:rtmp-rtsp-stream-client-java:samsungaudio-SNAPSHOT'

```
